File: backend/app/database/models/terminals/Terminal.py
import uuid
from sqlalchemy import Column, String, UUID
from sqlalchemy.dialects.postgresql import UUID

from ...base_model import BASE_MODEL


class Terminal(BASE_MODEL):
    __tablename__= "terminals"
    
    id = Column(UUID(as_uuid=True), default=uuid.uuid4, primary_key=True, index=True)
    name = Column(String(256), nullable=False)
    time_zone = Column(String(64), nullable=False)
    address = Column(String(256), nullable=True, default="")
    short_name = Column(String(64), nullable=True, default="")
    account_code = Column(String(64), nullable=True, default="")

File: frontend/demo-app/src/domains/terminals/types/terminal.ts
/**
 * Terminal types
 */

import { BaseEntity, Nullable } from '../../../shared/types/common';

export interface Terminal extends BaseEntity {
  name: string;
  address: Nullable<string>;
  time_zone: string;
  short_name: Nullable<string>;
  account_code: Nullable<string>;
}

export type CreateTerminalRequest = Omit<Terminal, keyof BaseEntity>;
export type UpdateTerminalRequest = Partial<CreateTerminalRequest>;
